content/en/blog/_posts/2023-03-08-memory-qos-cgroups-v2/index.md

```diff
@@ -0,0 +1,260 @@
+---
+layout: blog
+title: 'Alpha Support for Quality-of-Service for Memory Resources'
+date: 2023-03-08
+slug: qos-memory-resources
+---
+
+**Authors:** Dixita Narang (Google)
+
+Kubernetes v1.27, released in April 2023, introduced changes to an alpha feature,
+Memory QoS, that improves memory management capabilites in Linux nodes. 
+
+Support for Memory QoS feature was initially added in Kubernetes v1.22, and later some 
+[limitations](https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/2570-memory-qos#reasons-for-changing-the-formula-of-memoryhigh-calculation-in-alpha-v127) 
+around the formula for calculating memory.high were identified. These limitations are 
+addressed in Kubernetes v1.27.
+
+## Background
+
+Kubernetes allows you to optionally specify how much of each resources a container needs
+in the Pod specification. The most common resources to specify are CPU and Memory. 
+
+For example, a Pod spec with container resources could look like as follows:
+```
+apiVersion: v1
+kind: Pod
+metadata:
+  name: example
+spec:
+  containers:
+  - name: nginx
+    resources:
+      requests:
+        memory: "64Mi"
+        cpu: "250m"
+      limits:
+        memory: "64Mi"
+        cpu: "500m"
+```
+
+* `spec.containers[].resources.requests`
+
+  When you specify the resource request for containers in a Pod, the [kube-scheduler](/docs/concepts/scheduling-eviction/kube-scheduler/#kube-scheduler)
+  uses this information to decide which node to place the Pod on. Each node has a 
+  maximum capacity for each of the resource types: the amount of CPU and memory it 
+  can provide for Pods. The scheduler ensures that, for each resource type, the sum
+  of the resource requests of the scheduled containers is less than the capacity of
+  the node.
+
+* `spec.containers[].resources.limits`
+
+  When you specify the resource limit for containers in a Pod, the kubelet enforces 
+  those limits so that the running containers are not allowed to use more of those 
+  resources than the limits you set.
+
+When a node with enough capacity to serve the requested resources has been identified, 
+the kubelet starts the containers. When the kubelet starts a container as a part of a Pod, 
+kubelet passes the container's requests and limits for CPU and memory to the container 
+runtime. On Linux, the container runtime typically configures kernel cgroups as follows:
+
+* CPU limit
```

Comment:
nit: I think this explanation is really great, but since the focus here is on memory qos, the CPU section is a bit irrelavent since this feature is only looking at memory. However, I think introducing these concepts and explaning this is still useful... hmm thoughts?